### PR TITLE
Combine the two helper functions that disable servers OOB

### LINKED ISSUE #1442: Refactor oob_delete_then
Better if it were not be a decorator, but took a function as parameter "then".  Callbacks would make more sense if they looked like `self.helper.oob_delete_then(rcs, group, num, then=scaling_policy.execute)`.

Better yet if we can remove it and just use `_test_scaling_after_oobd`

### PR DESCRIPTION
**Pull request by cyli**
This also ensures checking the final state of the group in several tests that weren't before.

This also uses a `then` parameter for the `_oob_disable_then` function - thanks @sam-falvo and @kivattik 

Fixes #1442.

### CODE REVIEW

**Comment by cyli** [comment]
jenkins please run integration tests

**Comment by cyli** [comment]
jenkins please run integration tests

**Comment by radix** [comment]
looks like this needs merged forward

**Comment by cyli** [comment]
@radix: will do, thanks. :)  Going to see if I can track down some test failures in master first.

**Comment by cyli** [comment]
jenkins please run integration tests

File: otter/integration/tests/test_convergence.py [hunk 1]
```diff
@@ -478,59 +405,50 @@ def _remove_metadata(self, ids, rcs):
...
+    :param int min_servers: The min entities for the scaling group - defaults
+        to zero.
+    :param int max_servers: The min entities for the scaling group - defaults
+        to 25.
+    :param int desired_servers: The initial desired capacity of the scaling
+        group.  If not provided, by default, the scaling group will just start
+        out with the minimum number of servers. If provided, immediately after
+        creation the group will be scaled to this number before any deletions
+        or other scaling occurs.
+
```
Comment (radix) [inline otter/integration/tests/test_convergence.py]: Need documentation for `final_servers` too

**Comment by radix** [comment]
Took me a while to get caught up here, but it looks good! I just have one minor docstring thing. LGTM after that.

**Pull request merged**